dart_io_mini_samples/example/sockets/using_serversockets_server.dart

```diff
@@ -8,12 +8,8 @@
 import 'dart:io';
 import 'dart:convert';
 
-main() {
-  ServerSocket.bind('127.0.0.1', 4041)
-    .then((serverSocket) {
-      print('connected');
-      serverSocket.listen((socket) {
-        socket.transform(UTF8.decoder).listen(print);
-      });
-    });
+main() async {
+  var serverSocket = await ServerSocket.bind('127.0.0.1', 4041);
+  print('connected');
+  await for (var socket in serverSocket) socket.transform(UTF8.decoder).listen(print);
```

Comment:
Did you look into getting rid of the second listen() here?

(It probably made the code longer, so keeping listen probably makes sense.)